src/Repository/OrderingRepository.php

```diff
@@ -0,0 +1,260 @@
+<?php
+/**
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * It is available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/afl-3.0.php
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this file
+ *
+ *  @author    Buckaroo.nl <plugins@buckaroo.nl>
+ *  @copyright Copyright (c) Buckaroo B.V.
+ *  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+namespace Buckaroo\PrestaShop\Src\Repository;
+
+use Buckaroo\PrestaShop\Src\Entity\BkCountries;
+use Buckaroo\PrestaShop\Src\Entity\BkOrdering;
+use Buckaroo\PrestaShop\Src\Entity\BkPaymentMethods;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\NonUniqueResultException;
+
+class OrderingRepository extends EntityRepository implements BkOrderingRepositoryInterface
+{
+    public function findOneByCountryId($country_id)
+    {
+        return $this->findOneBy(['country_id' => $country_id]);
+    }
+
+    /**
+     * @throws NonUniqueResultException
+     */
+    public function findOneByCountryIsoCode(?string $isoCode2)
+    {
+        $qb = $this->_em->createQueryBuilder()->select('bo')
+            ->from(BkOrdering::class, 'bo');
+        if ($isoCode2 !== null) {
+            $qb->join(BkCountries::class, 'c', 'WITH', 'bo.country_id = c.country_id')
+                ->where('c.iso_code_2 = :isoCode2')
+                ->setParameter('isoCode2', $isoCode2);
+        } else {
+            $qb->where('bo.country_id IS NULL');
+        }
+
+        return $qb->getQuery()->getOneOrNullResult();
+    }
+
+    public function updateOrdering($value, $countryId = null)
+    {
+        if (is_string($value)) {
+            $value = json_decode($value, true);
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                // Handle JSON decode error
+                return false;
+            }
+        }
+
+        $idArray = array_column($value, 'id');
+        $value = json_encode($idArray);
+
+        if ($countryId !== null) {
+            return $this->updateOrderingForSpecificCountry($value, $countryId);
+        }
+
+        return $this->updateOrderingForNullCountry($value);
+    }
+
+    private function updateOrderingForSpecificCountry($value, $countryId)
+    {
+        $countryRepo = $this->_em->getRepository(BkCountries::class);
+        $country = $countryRepo->findOneBy(['country_id' => $countryId]);
+
+        if (!$country) {
+            return false;
+        }
+
+        $orderingRepo = $this->_em->getRepository(BkOrdering::class);
+        $ordering = $orderingRepo->findOneBy(['country_id' => $country->getCountryId()]);
+
+        if (!$ordering) {
+            return false;
+        }
+
+        $ordering->setValue($value);
+        $this->_em->persist($ordering);
+        $this->_em->flush();
+
+        return true;
+    }
+
+    private function updateOrderingForNullCountry($value)
```

Comment:
these code seems to repeat with the code from the method above, remove the duplicate code